subsys/shell/shell.c

```diff
@@ -1320,35 +1344,20 @@ void shell_fprintf(const struct shell *shell, enum shell_vt100_color color,
 
```

Comment:
you could add assert here. Something like: ASSERT(in_context || current_thread != shell_thread)

Reply:
done